### PR TITLE
Removed Unused Variables

### LINKED ISSUE #241: [eslint] remove non used vars
**What is the issue?**
There are 272 non used vars in the code .
They could not be cleaned by running `yarn lint --fix` or `npx lint -- --fix"

```
yarn lint
yarn run v1.22.5
$ eslint .

/home/anan/work/opensearch-js/index.js
  289:36  error  'reject' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/lib/Connection.js
  252:20  error  'depth' is defined but never used                   no-unused-vars
  252:27  error  'options' is defined but never used                 no-unused-vars
  253:13  error  'authorization' is assigned a value but never used  no-unused-vars
  268:13  error  'authorization' is assigned a value but never used  no-unused-vars

/home/anan/work/opensearch-js/lib/Helpers.js
  100:13  error  'method' is assigned a value but never used  no-unused-vars
  100:21  error  'body' is assigned a value but never used    no-unused-vars
  100:27  error  'index' is assigned a value but never used   no-unused-vars
  202:12  error  'size' is defined but never used             no-unused-vars
  335:38  error  'reject' is defined but never used           no-unused-vars
  349:40  error  'reject' is defined but never used           no-unused-vars
  652:40  error  'reject' is defined but never used           no-unused-vars

/home/anan/work/opensearch-js/lib/Transport.js
  674:28  error  'params' is defined but never used   no-unused-vars
  674:36  error  'options' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/scripts/release-canary.js
  93:18  error  'question' is defined but never used  no-unused-vars
  94:32  error  'reject' is defined but never used    no-unused-vars

/home/anan/work/opensearch-js/scripts/utils/generateApis.js
  270:30  error  'param' is defined but never used  no-unused-vars
  322:50  error  'index' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/acceptance/events-order.test.js
   60:41  error  'request' is defined but never used  no-unused-vars
   65:35  error  'request' is defined but never used  no-unused-vars
   70:43  error  'request' is defined but never used  no-unused-vars
   75:36  error  'request' is defined but never used  no-unused-vars
   80:21  error  'result' is defined but never used   no-unused-vars
   97:41  error  'request' is defined but never used  no-unused-vars
  102:35  error  'request' is defined but never used  no-unused-vars
  107:38  error  '_err' is defined but never used     no-unused-vars
  107:44  error  'request' is defined but never used  no-unused-vars
  111:36  error  'request' is defined but never used  no-unused-vars
  116:21  error  'result' is defined but never used   no-unused-vars
  133:41  error  'request' is defined but never used  no-unused-vars
  138:35  error  'request' is defined but never used  no-unused-vars
  143:38  error  '_err' is defined but never used     no-unused-vars
  143:44  error  'request' is defined but never used  no-unused-vars
  147:36  error  'request' is defined but never used  no-unused-vars
  152:21  error  'result' is defined but never used   no-unused-vars
  169:41  error  'request' is defined but never used  no-unused-vars
  174:35  error  'request' is defined but never used  no-unused-vars
  179:38  error  '_err' is defined but never used     no-unused-vars
  179:44  error  'request' is defined but never used  no-unused-vars
  183:36  error  'request' is defined but never used  no-unused-vars
  188:37  error  'result' is defined but never used   no-unused-vars
  200:15  error  'params' is defined but never used   no-unused-vars
  216:41  error  'request' is defined but never used  no-unused-vars
  221:35  error  'request' is defined but never used  no-unused-vars
  226:43  error  'request' is defined but never used  no-unused-vars
  231:36  error  'request' is defined but never used  no-unused-vars
  236:21  error  'result' is defined but never used   no-unused-vars
  246:15  error  'params' is defined but never used   no-unused-vars
  269:41  error  'request' is defined but never used  no-unused-vars
  274:35  error  'request' is defined but never used  no-unused-vars
  279:43  error  'request' is defined but never used  no-unused-vars
  284:36  error  'request' is defined but never used  no-unused-vars
  289:21  error  'result' is defined but never used   no-unused-vars
  306:41  error  'request' is defined but never used  no-unused-vars
  311:35  error  'request' is defined but never used  no-unused-vars
  316:38  error  '_err' is defined but never used     no-unused-vars
  316:44  error  'request' is defined but never used  no-unused-vars
  320:31  error  '_err' is defined but never used     no-unused-vars
  320:37  error  'request' is defined but never used  no-unused-vars
  326:47  error  'result' is defined but never used   no-unused-vars
  360:41  error  'request' is defined but never used  no-unused-vars
  365:35  error  'request' is defined but never used  no-unused-vars
  370:43  error  'request' is defined but never used  no-unused-vars
  375:36  error  'request' is defined but never used  no-unused-vars
  380:21  error  'result' is defined but never used   no-unused-vars
  411:43  error  'request' is defined but never used  no-unused-vars
  416:37  error  'request' is defined but never used  no-unused-vars
  421:45  error  'request' is defined but never used  no-unused-vars
  426:38  error  'request' is defined but never used  no-unused-vars
  431:23  error  'result' is defined but never used   no-unused-vars

/home/anan/work/opensearch-js/test/acceptance/resurrect.test.js
   85:25  error  'result' is defined but never used  no-unused-vars
   93:25  error  'result' is defined but never used  no-unused-vars
  149:25  error  'result' is defined but never used  no-unused-vars
  157:25  error  'result' is defined but never used  no-unused-vars
  165:25  error  'result' is defined but never used  no-unused-vars
  214:25  error  'result' is defined but never used  no-unused-vars
  222:25  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/acceptance/sniff.test.js
  202:36  error  'kill' is defined but never used    no-unused-vars
  223:36  error  'kill' is defined but never used    no-unused-vars
  251:36  error  'kill' is defined but never used    no-unused-vars
  279:23  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/benchmarks/suite.js
  128:35  error  'q' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/bundlers/parcel-test/index.js
  16:19  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/bundlers/rollup-test/index.js
  16:19  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/bundlers/webpack-test/index.js
  16:19  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/integration/helpers-secure/search.test.js
  40:16  error  'doc' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/integration/helpers/bulk.test.js
   60:12  error  'doc' is defined but never used  no-unused-vars
   63:16  error  'doc' is defined but never used  no-unused-vars
   88:12  error  'doc' is defined but never used  no-unused-vars
  162:12  error  'doc' is defined but never used  no-unused-vars
  191:12  error  'doc' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/integration/helpers/msearch.test.js
  52:16  error  'doc' is defined but never used     no-unused-vars
  93:75  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/integration/helpers/scroll.test.js
  52:16  error  'doc' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/integration/helpers/search.test.js
  52:16  error  'doc' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/unit/api.test.js
  109:15  error  'body' is defined but never used      no-unused-vars
  309:15  error  'body' is defined but never used      no-unused-vars
  309:21  error  'warnings' is defined but never used  no-unused-vars
  328:13  error  'body' is defined but never used      no-unused-vars
  356:72  error  'result' is defined but never used    no-unused-vars

/home/anan/work/opensearch-js/test/unit/child.test.js
   64:23  error  'res' is defined but never used   no-unused-vars
   66:24  error  'res' is defined but never used   no-unused-vars
   88:23  error  'res' is defined but never used   no-unused-vars
   90:24  error  'res' is defined but never used   no-unused-vars
  159:33  error  'meta' is defined but never used  no-unused-vars
  163:22  error  'res' is defined but never used   no-unused-vars
  178:33  error  'meta' is defined but never used  no-unused-vars
  182:27  error  'res' is defined but never used   no-unused-vars
  196:32  error  'meta' is defined but never used  no-unused-vars
  200:22  error  'res' is defined but never used   no-unused-vars
  215:32  error  'meta' is defined but never used  no-unused-vars
  219:27  error  'res' is defined but never used   no-unused-vars
  313:23  error  'res' is defined but never used   no-unused-vars
  315:24  error  'res' is defined but never used   no-unused-vars

/home/anan/work/opensearch-js/test/unit/client.test.js
   620:45  error  'result' is defined but never used              no-unused-vars
   620:53  error  'ConfigurationError' is defined but never used  no-unused-vars
   631:32  error  'makeRequest' is defined but never used         no-unused-vars
   631:45  error  'result' is defined but never used              no-unused-vars
   631:53  error  'ConfigurationError' is defined but never used  no-unused-vars
   647:32  error  'makeRequest' is defined but never used         no-unused-vars
   647:45  error  'result' is defined but never used              no-unused-vars
   647:53  error  'ConfigurationError' is defined but never used  no-unused-vars
   648:15  error  'params' is defined but never used              no-unused-vars
   648:23  error  'options' is defined but never used             no-unused-vars
   649:38  error  'reject' is defined but never used              no-unused-vars
   898:25  error  'body' is defined but never used                no-unused-vars
   924:25  error  'body' is defined but never used                no-unused-vars
  1076:21  error  'result' is defined but never used              no-unused-vars
  1105:21  error  'result' is defined but never used              no-unused-vars

/home/anan/work/opensearch-js/test/unit/connection.test.js
   292:13  error  'res' is defined but never used       no-unused-vars
   319:15  error  'res' is defined but never used       no-unused-vars
   345:15  error  'res' is defined but never used       no-unused-vars
   382:13  error  'res' is defined but never used       no-unused-vars
   416:13  error  'res' is defined but never used       no-unused-vars
   450:13  error  'res' is defined but never used       no-unused-vars
   561:13  error  'res' is defined but never used       no-unused-vars
   587:13  error  'res' is defined but never used       no-unused-vars
   619:13  error  'res' is defined but never used       no-unused-vars
   706:11  error  'res' is defined but never used       no-unused-vars
   892:20  error  'req' is defined but never used       no-unused-vars
   892:25  error  'res' is defined but never used       no-unused-vars
   905:13  error  'res' is defined but never used       no-unused-vars
   931:13  error  'res' is defined but never used       no-unused-vars
   986:10  error  'size' is defined but never used      no-unused-vars
  1000:11  error  'response' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/unit/events.test.js
   83:11  error  'result' is defined but never used  no-unused-vars
  131:11  error  'result' is defined but never used  no-unused-vars
  141:11  error  'result' is defined but never used  no-unused-vars
  192:11  error  'result' is defined but never used  no-unused-vars
  202:11  error  'result' is defined but never used  no-unused-vars
  253:11  error  'result' is defined but never used  no-unused-vars
  307:11  error  'result' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/unit/helpers/bulk.test.js
    76:20  error  'doc' is defined but never used     no-unused-vars
    81:16  error  'doc' is defined but never used     no-unused-vars
   119:20  error  'doc' is defined but never used     no-unused-vars
   124:16  error  'doc' is defined but never used     no-unused-vars
   158:20  error  'doc' is defined but never used     no-unused-vars
   163:16  error  'doc' is defined but never used     no-unused-vars
   206:20  error  'doc' is defined but never used     no-unused-vars
   251:20  error  'doc' is defined but never used     no-unused-vars
   291:20  error  'doc' is defined but never used     no-unused-vars
   299:16  error  'doc' is defined but never used     no-unused-vars
   362:20  error  'doc' is defined but never used     no-unused-vars
   413:20  error  'doc' is defined but never used     no-unused-vars
   488:20  error  'doc' is defined but never used     no-unused-vars
   518:19  error  'params' is defined but never used  no-unused-vars
   534:20  error  'doc' is defined but never used     no-unused-vars
   539:16  error  'doc' is defined but never used     no-unused-vars
   554:19  error  'params' is defined but never used  no-unused-vars
   570:20  error  'doc' is defined but never used     no-unused-vars
   575:16  error  'doc' is defined but never used     no-unused-vars
   635:20  error  'doc' is defined but never used     no-unused-vars
   640:16  error  'doc' is defined but never used     no-unused-vars
   661:19  error  'params' is defined but never used  no-unused-vars
   675:22  error  'doc' is defined but never used     no-unused-vars
   718:20  error  'doc' is defined but never used     no-unused-vars
   726:16  error  'doc' is defined but never used     no-unused-vars
   775:20  error  'doc' is defined but never used     no-unused-vars
   780:16  error  'doc' is defined but never used     no-unused-vars
   824:18  error  'doc' is defined but never used     no-unused-vars
   832:14  error  'doc' is defined but never used     no-unused-vars
   873:18  error  'doc' is defined but never used     no-unused-vars
   886:14  error  'doc' is defined but never used     no-unused-vars
   924:18  error  'doc' is defined but never used     no-unused-vars
   934:14  error  'doc' is defined but never used     no-unused-vars
   972:18  error  'doc' is defined but never used     no-unused-vars
   985:14  error  'doc' is defined but never used     no-unused-vars
  1026:18  error  'doc' is defined but never used     no-unused-vars
  1034:14  error  'doc' is defined but never used     no-unused-vars
  1097:18  error  'doc' is defined but never used     no-unused-vars
  1164:20  error  'doc' is defined but never used     no-unused-vars
  1167:16  error  'doc' is defined but never used     no-unused-vars
  1202:20  error  'doc' is defined but never used     no-unused-vars
  1223:20  error  'doc' is defined but never used     no-unused-vars
  1283:18  error  'doc' is defined but never used     no-unused-vars
  1288:14  error  'doc' is defined but never used     no-unused-vars
  1341:18  error  'doc' is defined but never used     no-unused-vars
  1346:14  error  'doc' is defined but never used     no-unused-vars
  1387:18  error  'doc' is defined but never used     no-unused-vars
  1392:14  error  'doc' is defined but never used     no-unused-vars

/home/anan/work/opensearch-js/test/unit/helpers/msearch.test.js
   40:15  error  'params' is defined but never used  no-unused-vars
   89:15  error  'params' is defined but never used  no-unused-vars
  165:15  error  'params' is defined but never used  no-unused-vars
  239:15  error  'params' is defined but never used  no-unused-vars
  266:15  error  'params' is defined but never used  no-unused-vars
  280:75  error  'result' is defined but never used  no-unused-vars
  289:15  error  'params' is defined but never used  no-unused-vars
  301:62  error  'result' is defined but never used  no-unused-vars
  316:15  error  'params' is defined but never used  no-unused-vars
  328:43  error  'result' is defined but never used  no-unused-vars
  342:15  error  'params' is defined but never used  no-unused-vars
  402:15  error  'params' is defined but never used  no-unused-vars
  436:15  error  'params' is defined but never used  no-unused-vars
  471:15  error  'params' is defined but never used  no-unused-vars
  487:6   error  'e' is defined but never used       no-unused-vars
  490:12  error  'e' is defined but never used       no-unused-vars
  497:15  error  'params' is defined but never used  no-unused-vars
  518:52  error  'result' is defined but never used  no-unused-vars
  527:15  error  'params' is defined but never used  no-unused-vars
  595:15  error  'params' is defined but never used  no-unused-vars
  651:15  error  'params' is defined but never used  no-unused-vars
  684:77  error  'result' is defined but never used  no-unused-vars
  696:15  error  'params' is defined but never used  no-unused-vars
  720:15  error  'params' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/unit/helpers/scroll.test.js
  194:15  error  'params' is defined but never used  no-unused-vars
  341:15  error  'params' is defined but never used  no-unused-vars

/home/anan/work/opensearch-js/test/unit/transport.test.js
   455:13  error  'body' is defined but never used        no-unused-vars
   487:13  error  'body' is defined but never used        no-unused-vars
   517:13  error  'body' is defined but never used        no-unused-vars
   547:13  error  'body' is defined but never used        no-unused-vars
   580:15  error  'body' is defined but never used        no-unused-vars
   620:13  error  'body' is defined but never used        no-unused-vars
   658:13  error  'body' is defined but never used        no-unused-vars
   764:15  error  'body' is defined but never used        no-unused-vars
   817:15  error  'body' is defined but never used        no-unused-vars
   873:15  error  'body' is defined but never used        no-unused-vars
   985:13  error  'result' is defined but never used      no-unused-vars
  1099:11  error  'result' is defined but never used      no-unused-vars
  1155:13  error  'result' is defined but never used      no-unused-vars
  1228:15  error  'body' is defined but never used        no-unused-vars
  1334:15  error  'body' is defined but never used        no-unused-vars
  1400:27  error  'hosts' is defined but never used       no-unused-vars
  1442:17  error  'body' is defined but never used        no-unused-vars
  1475:18  error  'connection' is defined but never used  no-unused-vars
  1549:13  error  'body' is defined but never used        no-unused-vars
  1562:13  error  'body' is defined but never used        no-unused-vars
  1600:15  error  'body' is defined but never used        no-unused-vars
  1643:19  error  'body' is defined but never used        no-unused-vars
  1680:19  error  'body' is defined but never used        no-unused-vars
  1718:19  error  'body' is defined but never used        no-unused-vars
  1755:19  error  'body' is defined but never used        no-unused-vars
  1850:15  error  'body' is defined but never used        no-unused-vars
  1879:15  error  'body' is defined but never used        no-unused-vars
  1969:15  error  'body' is defined but never used        no-unused-vars
  2364:17  error  'body' is defined but never used        no-unused-vars
  2372:21  error  'body' is defined but never used        no-unused-vars
  2380:25  error  'body' is defined but never used        no-unused-vars
  2445:23  error  'meta' is defined but never used        no-unused-vars
  2750:15  error  'body' is defined but never used        no-unused-vars
  2814:17  error  'body' is defined but never used        no-unused-vars
  2850:17  error  'body' is defined but never used        no-unused-vars
  2922:13  error  'body' is defined but never used        no-unused-vars
  2957:13  error  'body' is defined but never used        no-unused-vars

/home/anan/work/opensearch-js/test/utils/buildProxy.js
  30:32  error  'reject' is defined but never used    no-unused-vars
  39:32  error  'reject' is defined but never used    no-unused-vars
  47:23  error  'handler' is defined but never used   no-unused-vars
  47:32  error  'callback' is defined but never used  no-unused-vars
  48:32  error  'reject' is defined but never used    no-unused-vars
  56:29  error  'handler' is defined but never used   no-unused-vars
  56:38  error  'callback' is defined but never used  no-unused-vars
  57:32  error  'reject' is defined but never used    no-unused-vars

/home/anan/work/opensearch-js/test/utils/buildServer.js
  68:34  error  'reject' is defined but never used  no-unused-vars

✖ 272 problems (272 errors, 0 warnings)

```

**How can one reproduce the bug?**
**After eslint PR is merged**
Go to '.eslintrc.json'
Remove ' "no-dupe-else-if": "off"'
Run yarn lint
See error

**What is the expected behavior?**
clean out

### PR DESCRIPTION
**Pull request by nhtruong**
### Description
Removed Unused Variables
- Removed `"no-unused-vars": "off"` in eslintrc.
- Disabled eslint's `no-unused-vars` check where the unused vars actually serve a purpose, and removed the unused vars otherwise.

### Issues Resolved
Closes #241 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by harshavamsi — APPROVED** [review]
I would have liked to maybe see un-used variables prepended with `_` and adding eslint rules to ignore them. Some of the variables removed here like `reject` can technically be used but are dormant in most cases. But I guess they could be re-added on a case-by-case basis. Otherwise LGTM!

**Review by VachaShah — APPROVED** [review]
Thank you @nhtruong!

File: lib/Connection.js [hunk 1]
```diff
@@ -249,7 +249,8 @@ class Connection {
   // We want to hide `auth`, `agent` and `ssl` since they made
   // the logs very hard to read. The user can still
   // access them with `instance.agent` and `instance.ssl`.
-  [inspect.custom](depth, options) {
+  [inspect.custom]() {
+    // eslint-disable-next-line no-unused-vars
     const { authorization, ...headers } = this.headers;
```
Comment (nhtruong) [inline lib/Connection.js:254]: Note to reviewer: Even though `authorization` is not being used, its declaration is still necessary to remove `authorization` from `headers`

File: lib/Connection.js [hunk 2]
```diff
@@ -265,6 +266,7 @@ class Connection {
   }
 
   toJSON() {
+    // eslint-disable-next-line no-unused-vars
     const { authorization, ...headers } = this.headers;
```
Comment (nhtruong) [inline lib/Connection.js:270]: Note to reviewer: Even though `authorization` is not being used, its declaration is still necessary to remove `authorization` from `headers`

**Comment by nhtruong** [comment]
@harshavamsi Good idea on the pattern to ignore underscored var names. I've updated eslintrc accordingly so that we can keep some unused variables around in some edge cases (like backward compatibility where some optional var is no longer used but still accepted to avoid breaking old code). For the cases like `reject`, we can always add it back when we need to use it.

**Comment by dblock** [comment]
@harshavamsi you're cool with merging this as is? or do we want to standardize on something like `_` instead of removing the unused args?

**Comment by harshavamsi** [comment]
> @harshavamsi you're cool with merging this as is? or do we want to standardize on something like `_` instead of removing the unused args?

I think @nhtruong has added a [rule](https://github.com/opensearch-project/opensearch-js/pull/301/files#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fR26) that ignores `_` prepended vars. I think this change is okay to go and newer variables with `_` can be ignored if necessary.

**Review by dblock — APPROVED** [review]

**Pull request merged**